Sukanta_Roy/progress.md

```diff
@@ -3,78 +3,63 @@
 **Project: Questions+**
 
 Overall Direction:
-- 
-- 
-- 
-
+- Rubric questions: allow weightages for sub questions
+- MCQ questions: support attaching weights to options
+- Calculate more statistics for MCQ questions (Total, average for each student)
+- MSQ questions: support attaching weights to options
+- Calculate more statistics for MSQ questions
+- Ensure the downloaded data contains all data shown in the page.
+- Add a new question template
 ---
 
 Week 1 Plan:
-- [ ] 
-- [ ] 
-- [ ] 
+- [ ] Implement the frontend part of [Rubric questions: allow weightages for sub questions](https://github.com/TEAMMATES/teammates/issues/7224)
+- [ ] Start Implementing the backend for weights feature in rubric questions sub questions
 
 Week 2 Plan:
-- [ ] 
-- [ ] 
-- [ ] 
+**Final Semester Exams**
+
+- [ ] Finalize the sample question list for the new question template
 
 Week 3 Plan:
-- [ ] 
-- [ ] 
-- [ ] 
+- [ ] Implement the backend for [Rubric questions: allow weightages for sub questions](https://github.com/TEAMMATES/teammates/issues/7224)
+- [ ] Update documentation for rubric question weight feature in the instructor help page
+- [ ] Implement the frontend part of [MCQ questions: support attaching weights to options](https://github.com/TEAMMATES/teammates/issues/2776)
 
 Week 4 Plan:
-- [ ] 
-- [ ] 
-- [ ] 
+- [ ] Implement the backend part of [MCQ questions: support attaching weights to options](https://github.com/TEAMMATES/teammates/issues/2776)
```

Comment:
I noticed implementing backends for MCQ and MSQ weights feature will take a whole week, but the weight feature backend for rubric sub questions takes less than that. Why is it so?

Reply:
@bqnguyen94 At first I thought for mcq and msq questions I'm adding a new feature from scratch and for rubric question I'll be extending the existing weights feature to be applied to each cell. I did not take into account the data migration aspect, so it is a mistake in my judgement. Rubric question's backend part will most probably also need a full week or (I hope not) more. Which is Why I'm revising my plan a little bit right now.